src/System.Security.Claims/src/System/Security/Claims/GenericPrincipal.cs

```diff
@@ -86,7 +124,7 @@ public override bool IsInRole(string role)
 
             for (int i = 0; i < m_roles.Length; ++i)
             {
-                if (m_roles[i] != null && String.Compare(m_roles[i], role, StringComparison.OrdinalIgnoreCase) == 0)
+                if (string.Equals(m_roles[i], role, StringComparison.OrdinalIgnoreCase))
```

Comment:
`role` was already confirmed to be not null. So if `m_roles[i]` is null it wont be equal.